File: Jan/Week 3/LeetCode/problem_5.py
# Day 19 Problem No 41

# 1732. Find the Highest Altitude => https://leetcode.com/problems/find-the-highest-altitude/?envType=study-plan-v2&envId=leetcode-75

class Solution:
    def largestAltitude(self, gain: List[int]) -> int:
        curr_alt = 0
        high_alt = curr_alt
        for x in gain:
            curr_alt+=x
            high_alt = max(high_alt,curr_alt)

        return high_alt
    

File: Jan/Week 4/LeetCode/Problem_5.cpp
// Day 25 Problem No 54

// 1456. Maximum Number of Vowels in a Substring of Given Length => https://leetcode.com/problems/maximum-number-of-vowels-in-a-substring-of-given-length/description/?envType=study-plan-v2&envId=leetcode-75

class Solution {
public:
    int maxVowels(string s, int k) {
        int mxcnt = 0;
        int currcnt = 0;
        auto isV = [](char c){
            return c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u';
        };
        for(int i=0;i<k;i++){
            if(isV(s[i])) currcnt++;
        }
        mxcnt = currcnt;
        for(int i=k;i<s.size();i++){
            if(isV(s[i])) currcnt++;
            if(isV(s[i-k])) currcnt--;
            mxcnt = max(mxcnt,currcnt);
        }
        return mxcnt;
    }
};